lib/rust/ensogl/core/src/animation/loops.rs

```diff
@@ -19,7 +19,7 @@ use web::Closure;
 /// differ across browsers and browser versions. We have even observed that `performance.now()` can
 /// sometimes provide a bigger value than time provided to `requestAnimationFrame` callback later,
 /// which resulted in a negative frame time.
-#[derive(Clone, Copy, Debug, Default)]
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq)]
```

Comment:
I just realized that `Duration` implements `Eq` but is based on `f32`. Will this not lead to issues where durations end up with `NaN`s.  Or is this accounted for? If not we should consider doing special handling for this.